### PR TITLE
🚀 release: sync dev/v1.6 → main for v1.6.0-rc.11

### LINKED ISSUE #632: feat(portwing): route remote watch and update triggers through EdgeAgentAdapter
## Problem

The 2026-07-28 Sockguard/Portwing/drydock compatibility audit found that the tri-tool stack supports discovery and state synchronization but cannot perform drydock-triggered remote updates through Portwing.

- `AgentClient.runRemoteTrigger`, `watch`, and `watchContainer` do not branch to the Edge adapter.
- `EdgeAgentAdapter` currently treats watcher/trigger responses as logging-only no-ops rather than completing the initiating operation.
- Standard Mode reaches Portwing endpoints that currently return 501.

Portwing-side contract work is tracked in CodesWhat/portwing#76.

## Proposed scope

- Add Edge adapter operations for single/batch remote triggers and watcher/container-watch requests.
- Correlate each request with a bounded pending operation and resolve/reject it from the agent response.
- Propagate disconnect, timeout, authorization, and agent errors to the caller.
- Preserve the existing Standard Mode path and feature-detect unsupported components.
- Add an end-to-end conformance test with Portwing for both Standard and Edge transports.
- Document that `serverCompatLevel` wire compatibility does not by itself imply watcher/trigger feature compatibility.

## Acceptance

- A Portwing-backed update invoked from drydock completes or returns a concrete error to the UI/API in both supported transports.
- No watcher/trigger response is silently discarded.
- Existing Edge reconnect, exec, log-stream, backpressure, and inventory tests remain green.

### LINKED ISSUE #637: Container lifecycle actions (start/stop/restart) 404 for edge-agent containers
`start`/`stop`/`restart` container actions return 404 "No docker trigger found" for every container tracked via a Portwing edge agent: those actions require a local `DD_ACTION_DOCKER_*` trigger and never route through `EdgeAgentAdapter`, which has no lifecycle methods (only discovery/watch state, continuous logs, and deleteContainer are wired; `dd:trigger_response` is a logged no-op).

This is the same architecture boundary as update-triggers-over-edge (the docs now call that out — see the docs/v1.6-portwing-edge-gaps PR), but lifecycle buttons in the UI fail with a confusing 404 rather than a capability message. Found live during the 2026-07-28 tri-tool NAS acceptance (item 4).

Two-part fix candidate: (a) short-term, surface a clear "not supported over edge agents" error/disabled state instead of the 404; (b) roadmap, extend the portwing/1.0 edge protocol with lifecycle calls alongside the planned watcher/trigger contract. Related: #635 (exec_end reason), the edge exec surface (unwired startExec).

### PR DESCRIPTION
**Pull request by scttbnsn**
Promotes the reviewed Portwing controller-owned Docker transport from dev/v1.6 to main for v1.6.0-rc.11. The promotion commit tree is byte-identical to the current dev/v1.6 tree: 567e23a05b4d2d6545cb721da80e7db75f131518. Companion Portwing change: https://github.com/CodesWhat/portwing/pull/81

Closes #632
Closes #637

## Changelog

✨ Added
- Added controller-owned Docker transport for Portwing watcher, container, update, preview, and rollback operations.
- Added `AgentClient.requestDockerApi()` and `DockerApiProxyResponse`.
- Added `PortwingDockerBridge` with authentication, request limits, header filtering, error mapping, and shutdown cleanup.
- Added controller transport marker validation.
- Added correlated Edge request handling for responses, disconnects, timeouts, authorization failures, and agent errors.
- Added ordered Edge state-frame processing and bounded streamed-response handling.
- Added Ed25519 signature version 2 with exact path and query-string canonicalization.
- Added centralized fleet-agent component cleanup.
- Added regression coverage for transport, lifecycle, signing, cleanup, ordering, and registry behavior.

🔧 Changed
- Preserve Standard HTTP behavior when controller Docker transport is unavailable.
- Preserve controller-owned container enrichment across watcher and update events.
- Preserve durable Portwing name bindings across restarts.
- Update release metadata and documentation to `1.6.0-rc.11`.

🐛 Fixed
- Prevent stale component cleanup from removing replacement registry entries.
- Prevent watcher and trigger responses from being silently discarded.
- Prevent oversized streamed responses and Docker bridge requests.
- Prevent cleanup failures from blocking remaining agent cleanup.
- Prevent signing mismatches for escaped targets, query strings, and request bodies.
- Handle aborted Portwing bridge requests.

## Concerns

- Verify unsupported controller lifecycle actions return explicit errors instead of 404 responses.
- Verify legacy and Docker Compose Portwing paths return the documented `501` response.
- Verify signature version 2 remains compatible with all Standard Mode clients.
- Verify every controller watcher and trigger registration has rollback and deregistration coverage.
- Verify response-size and request-body limits match deployment requirements.
- Verify release-version updates remain consistent across runtime data, mocks, documentation, and tests.

### CODE REVIEW

File: app/agent/EdgeAgentAdapter.ts [hunk 1]
```diff
@@ -1296,7 +1341,15 @@ export class EdgeAgentAdapter {
...
+      const componentCleanup = registry
+        .deregisterAgentComponents(this.agentName)
+        .catch((error: unknown) => {
+          log.warn(
+            `Failed to clean up components for disconnected edge agent ${this.agentName}: ${getErrorMessage(error)}`,
+          );
+        });
       removeAgent(this.agentName);
+      await componentCleanup;
     }
```
[the remark this answers is not in the document]
Comment (scttbnsn) [reply app/agent/EdgeAgentAdapter.ts:1353]: Verified against the actual event-loop and registry ordering; no code change is needed. onDisconnect calls deregisterAgentComponents synchronously before removeAgent. That async function snapshots the old watcher and trigger object references before its first await, so a same-name replacement cannot register before the snapshot. If old deregistration completes after the replacement registers, deregisterComponent deletes only when the registry entry is still the exact snapshotted object instance. The existing registry regression test named late cleanup of a disconnected agent does not delete a replacement component with the same id covers this identity guard, and the Edge regression covers stale-adapter same-name reconnect eviction. Adding a getAgent recheck after removeAgent would skip and leak the old cleanup, while a recheck before the snapshot would be redundant because no yield occurs.

File: app/agent/PortwingDockerBridge.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    this.maxRequestBodyBytes = options.maxRequestBodyBytes ?? DEFAULT_MAX_REQUEST_BODY_BYTES;
+  }
+
+  async start(): Promise<PortwingDockerBridgeEndpoint> {
+    if (this.endpoint) {
+      return this.endpoint;
+    }
+    const server = http.createServer((req, res) => {
+      void this.handle(req, res);
+    });
```
[the remark this answers is not in the document]
Comment (scttbnsn) [reply app/agent/PortwingDockerBridge.ts:90]: Confirmed and reproduced with a real authenticated partial POST on Node 24: aborting the body produced an unhandled ECONNRESET rejection and did not call the upstream requester. The test-first fix is in #652 against dev/v1.6. It attaches a rejection handler at the HTTP server boundary and destroys only the affected response. The new regression failed on the old code and passes with the fix; the full local pre-push gate passed, including exact 100% app/UI coverage. I will update the release-sync PR from the merged dev tree before resolving this thread.

**Comment by scttbnsn** [comment]
Ready for approval: every required check is green, CodeRabbit generated no actionable comments on the latest head, and all review threads are resolved. @ALARGECOMPANY @biggest-littlest — this PR now only needs the two required approvals, including CODEOWNER approval. Squash auto-merge is enabled and will preserve the reviewed tree.

**Review by ALARGECOMPANY — APPROVED** [review]
Reviewed after all required CI and CodeRabbit checks completed successfully. Approving the latest head for release.

**Review by biggest-littlest — APPROVED** [review]
Reviewed the coordinated Drydock release promotion and successful automated verification. Approving the latest head.
